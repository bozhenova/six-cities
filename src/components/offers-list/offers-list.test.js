import React from 'react';
import OffersList from './offers-list';
import renderer from 'react-test-renderer';

describe('Offers List', () => {
  const props = {
    handleCardHover: () => {},
    offers: [
      {
        title: '',
        previewPhoto: '',
        isPremium: true,
        isFavorite: true,
        price: 116,
        type: '',
        rating: 80
      },
      {
        title: '',
        previewPhoto: '',
        isPremium: true,
        isFavorite: false,
        price: 136,
        type: '',
        rating: 88
      }
    ]
  };

  it('should render a list of offers', () => {
    const tree = renderer.create(<OffersList {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
