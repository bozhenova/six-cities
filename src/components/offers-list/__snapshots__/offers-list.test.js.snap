// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offers List should render a list of offers 1`] = `
<ul
  className="cities__places-list places__list tabs__content"
>
  <li>
    <article
      className="cities__place-card place-card"
      onMouseEnter={[Function]}
    >
      <div
        className="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src=""
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              116
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button place-card__bookmark-button--active"
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            className="title"
            href="#"
          >
            
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          
        </p>
      </div>
    </article>
  </li>
  <li>
    <article
      className="cities__place-card place-card"
      onMouseEnter={[Function]}
    >
      <div
        className="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src=""
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              136
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button"
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "88%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            className="title"
            href="#"
          >
            
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          
        </p>
      </div>
    </article>
  </li>
</ul>
`;
